ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -164,7 +164,7 @@ public void run() {
 
           for (CompactionInfo ci : potentials) {
             try {
-              Table t = resolveTable(ci);
+              Table t = resolveTableAndCache(ci);
```

Comment:
Issue is more pronounced in the following code path.

{code}
Set<CompactionInfo> potentials = compactionExecutor.submit(() ->
            txnHandler.findPotentialCompactions(abortedThreshold, abortedTimeThreshold, compactionInterval)
              .parallelStream()
              .filter(ci -> isEligibleForCompaction(ci, currentCompactions, skipDBs, skipTables))
              .collect(Collectors.toSet())).get();
{code}

i.e in `isEligibleForCompaction`.  

Instead of introducing the cache and invalidating it later, is it possible to create a simple hashmap in the method itself and use it on need basis? (e.g if there are multiple partitions, it will just do one lookup for table and the rest of the partitions will make use of the value in map).

Reply:
The idea was, that this way we can have a shared cache between the initiator and the cleaner.

Reply:
1. In large clusters, there can be 10000s of tables and having additional cache could add up to mem pressure. 
2. As mentioned earlier, issue was observed more in isEligibleForCompaction. It would be helpful if the patch can address it.

Reply:
I'm fine changing it to the suggested approach, just want to understand the reasoning behind it. 
I do have concerns introducing a hash map, as we have less control over the memory pressure. Table objects are moderately large, and as you mentioned there can be large clusters with numerous tables.
I was using softValues, so the cache can be cleared by the garbage collector in response to memory demand.
As for no 2, do you see some use case where the current approach doesn't address the observed issue?